sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveSessionStateBuilder.scala

```diff
@@ -42,7 +42,7 @@ class HiveSessionStateBuilder(session: SparkSession, parentState: Option[Session
    * Create a Hive aware resource loader.
    */
   override protected lazy val resourceLoader: HiveSessionResourceLoader = {
-    val client: HiveClient = externalCatalog.client.newSession()
+    val client: HiveClient = externalCatalog.client
```

Comment:
newSession is to isolate SparkSession in Spark ThriftServer, You must judge whether is thriftserver or not.